Add guidance for backups to NFS share (#3081)
theforeman/foreman-documentation: guides/common/modules/proc_performing-a-full-backup.adoc

You can use both online and offline methods to back up remote databases, but if you use offline method, the `{foreman-maintain} backup` command performs a database dump.

.Backing up to a remote NFS share

> To enable {Project} to save the backup to the NFS share, make sure that the `root` user of your {ProjectServer} or {SmartProxyServer} can write to the NFS share.

Reviewer: ```suggestion
To enable {Project} to save the backup to an NFS share, make sure that the `root` user of your {ProjectServer} or {SmartProxyServer} can write to the NFS share.
```

Not sure about `a`/`an`, but there is no specific NFS share at that point, so `the` seems wrong :)

You can use both online and offline methods to back up remote databases, but if you use offline method, the `{foreman-maintain} backup` command performs a database dump.

.Backing up to a remote NFS share

> 

Reviewer 2: ```suggestion
```

> To enable {Project} to save the backup to an NFS share, make sure that the `root` user of your {ProjectServer} or {SmartProxyServer} can write to the NFS share.

Reviewer 2: ```suggestion
To enable {Project} to save the backup to an NFS share, ensure that the `root` user of your {ProjectServer} or {SmartProxyServer} can write to the NFS share.
```

You can use both online and offline methods to back up remote databases, but if you use offline method, the `{foreman-maintain} backup` command performs a database dump.

.Backing up to a remote NFS share

You can back up your {ProjectServer} or {SmartProxyServer} in a remote NFS share.

> * Make sure you export the share in your NFS server with the `no_root_squash` option to disable root squashing.

Reviewer: I think that is bad advise, even if we add that warning at the end. How about:
```suggestion
* Make sure the `root` user of your {ProjectServer} or {SmartProxyServer} can write to the NFS share.
NFS export options like `root_squash` and `all_squash` are known to prevent this.
For more information, see https://docs.redhat.com/en/documentation/red_hat_enterprise_linux/9/html/configuring_and_using_network_file_services/deploying-an-nfs-server_configuring-and-using-network-file-services
```

Especially, you can backup to `root_squash`ed NFS share just fine if the directory you write to is writable by `nobody` on the NFS server (and you use `--preserve-directory` to foreman-maintain, but that part is a bug in f-m that I am going to fix).

Reviewer: https://github.com/theforeman/foreman_maintain/pull/888 for the code change

Reviewer 3: @asteflova I would say, 
* Have QE do some testing with some recommended options
* The test should be done for both backup as well as restore
* Once done and confirmed to be successfully happening, Document those options as primary recommendations

An end-user can export the NFS share with different combinations of options. So the best we can do is to provide:
* Minimum recommended options that would work
* Some NFS export options that may affect the functionality and should not be used

Reviewer: Well, to be honest, my whole point is that we should NOT document ANY NFS options as we're not managing the NFS server. Especially as those options vary in meaning depending on the NFS implementation.

E.g. I have a NetApp NFS here, which I assume has `root_squash` on, as any file I create as `root` gets `nobody` as owner instead, but I can backup there just fine as `chown .postgres` works. When I replicate the setup with an EL9 NFS server, I get "Operation not permitted" for the `chown`.

Reviewer 3: Without any recommendations at all, The end-users run into permission errors and come back to RH for further support on resolving those matters. So if we mention something like NFS share can be used, we also should mention how they can confirm that the NFS share is usable.

So if not recommendations about exporting NFS share, We can document some points about how to check\test whether the exported share is usable for a backup\restore operation or not.

Reviewer: > So if we mention something like NFS share can be used, we also should mention how they can confirm that the NFS share is usable.

That's why I wrote "Make sure the `root` user of your can write to the NFS share." as that's exactly the test for usability of the share. Which btw also applies to any other storage, not only NFS (it's just for local permission checks `root` usually has `DAC_OVERRIDE` capability and can ignore any constraints).

Author: Applied.

.Backing up to a remote NFS share

To enable {Project} to save the backup to the NFS share, make sure that the `root` user of your {ProjectServer} or {SmartProxyServer} can write to the NFS share.
NFS export options such as `root_squash` and `all_squash` are known to prevent this.

> For more information, see link:{RHELDocsBaseURL}9/html/configuring_and_using_network_file_services/deploying-an-nfs-server_configuring-and-using-network-file-services[{RHEL} _Configuring and using network files services_] and link:{RHELDocsBaseURL}9/html/securing_networks/securing-network-services_securing-networks#export-options-for-securing-an-nfs-server_securing-the-nfs-service[{RHEL} _Securing network services_].

Reviewer 2: ```suggestion
ifndef::orcharhino[]
For more information, see link:{RHELDocsBaseURL}9/html/configuring_and_using_network_file_services/deploying-an-nfs-server_configuring-and-using-network-file-services[{RHEL} _Configuring and using network files services_] and link:{RHELDocsBaseURL}9/html/securing_networks/securing-network-services_securing-networks#export-options-for-securing-an-nfs-server_securing-the-nfs-service[{RHEL} _Securing network services_].
endif::[]
```
